vm/interpreter/src/default_syscalls.rs

```diff
@@ -0,0 +1,201 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::resolve_to_key_addr;
+use actor::miner;
+use blocks::BlockHeader;
+use cid::Cid;
+use clock::ChainEpoch;
+use forest_encoding::from_slice;
+use ipld_blockstore::BlockStore;
+use runtime::{ConsensusFault, ConsensusFaultType, Syscalls};
+use state_tree::StateTree;
+use vm::{ActorError, ExitCode};
+
+/// Default syscalls information
+#[derive(Clone, Debug)]
+pub struct DefaultSyscalls<'bs, BS> {
+    store: &'bs BS,
+}
+
+impl<'bs, BS> DefaultSyscalls<'bs, BS> {
+    /// DefaultSyscalls constuctor
+    pub fn new(store: &'bs BS) -> Self {
+        Self { store }
+    }
+}
+
+impl<'bs, BS> Syscalls for DefaultSyscalls<'bs, BS>
+where
+    BS: BlockStore,
+{
+    /// Verifies that two block headers provide proof of a consensus fault:
+    /// - both headers mined by the same actor
+    /// - headers are different
+    /// - first header is of the same or lower epoch as the second
+    /// - at least one of the headers appears in the current chain at or after epoch `earliest`
+    /// - the headers provide evidence of a fault (see the spec for the different fault types).
+    /// The parameters are all serialized block headers. The third "extra" parameter is consulted only for
+    /// the "parent grinding fault", in which case it must be the sibling of h1 (same parent tipset) and one of the
+    /// blocks in the parent of h2 (i.e. h2's grandparent).
+    /// Returns an error if the headers don't prove a fault.
+    fn verify_consensus_fault(
+        &self,
+        h1: &[u8],
+        h2: &[u8],
+        extra: &[u8],
+        _earliest: ChainEpoch, // unused in lotus
+    ) -> Result<Option<ConsensusFault>, ActorError> {
```

Comment:
Looking at specs-actor, the have `ConsensusFaultNone = 0` commented out. I wonder if they're gunna uncomment that so that we dont need to return an Option.

Reply:
They probably just had that because they didn't want to use pointers/ nil for safety, technically better if they don't and just handle